src/asmcnc/apps/drywall_cutter_app/material_setup_popup.py

```diff
@@ -0,0 +1,516 @@
+                self.update_graphic_position()
+                self.calculate_depth_per_pass(self.depth_per_pass, self.auto_pass_checkbox.active)
+
+                if float(self.total_cut_depth.text) > max_cut_depth:
+                    self.float_layout.add_widget(self.cut_depth_warning)
+                elif float(self.total_cut_depth.text) <= 0:
+                    self.cut_depth_warning.text = self.cut_depth_warning_zero
+                    self.float_layout.add_widget(self.cut_depth_warning)
+                else:
+                    self.float_layout.remove_widget(self.cut_depth_warning)
+
+                self.disable_confirm_button()
+            except:
+                pass
+
+    def disable_confirm_button(self):
+        children = self.float_layout.children
+        if self.cut_depth_warning in children or self.pass_depth_warning in children:
+            self.confirm_button.disabled = True
+            self.confirm_button.opacity = 0.5
+        else:
+            self.confirm_button.disabled = False
+            self.confirm_button.opacity = 1
+        self.float_layout.do_layout()
+
+    def generate_pass_depth_lines(self, number_of_passes):
+        upper_limit = 0.0825  # Value at which the line sits at the top of the material
+        lower_limit = -0.175  # Value at which the line sits at the bottom surface of the material
+        line_range = upper_limit - lower_limit
+        material_thickness = 0 if self.material_thickness.text == '' or self.material_thickness.text == '-' else float(
+            self.material_thickness.text)
+        depth_per_pass = 0 if self.depth_per_pass.text == '' or self.depth_per_pass.text == '-' else float(
+            self.depth_per_pass.text)
+        ratio = 0 if material_thickness == 0 else depth_per_pass / material_thickness
+        singular_cut = 0 if self.depth_per_pass.text == "" else round(ratio * line_range, 4)
+        for pass_line in self.pass_depth_lines:
+            self.cutter_layout.remove_widget(pass_line)
+        if self.total_cut_depth.text != "0.0":
+            for i in range(0, int(number_of_passes)):
+                img = Image()
+
+                self.cutter_layout.add_widget(img)
+
+                img.source = "./asmcnc/apps/drywall_cutter_app/img/pass_depth_line.png"
+
+                line_depth = upper_limit - (singular_cut * (i+1))
+                cutter_limit = lower_limit - (-0.035 - self.cutter_graphic.pos_hint['y'])
+                if line_depth < cutter_limit:
+                    line_depth = cutter_limit
+
+                img.pos_hint = {'center_x': 0.45, 'y': line_depth}
+                img.allow_stretch = True
+
+                self.pass_depth_lines.append(img)
+        self.cutter_layout.do_layout()
+
+    def calculate_depth_per_pass(self, *args):
+        max_cut_depth_per_pass = self.dwt_config.active_cutter.max_depth_per_pass
+        if self.auto_pass_checkbox.active:
+            try:
```

Comment:
again, not sure what the try: except is doing

if it's about whether you can float(self.total_cut_depth.text), check this first, if you can't then you can just return. makes the flow easier to see.